dali/operators/math/normalize/normalize.cc

```diff
@@ -301,29 +301,29 @@ void Normalize<CPUBackend>::RunTyped(HostWorkspace &ws) {
   if (batch_norm_) {
     if (ShouldCalcMean()) {
       for (int i = 0; i < nsamples; i++) {
-        tp.DoWorkWithID([&, i](int thread_idx) {
+        tp.AddWork([&, i](int thread_idx) {
           kernels::MeanCPU<float, InputType> mean;
           mean.Setup(mutable_mean[i], in_view[i], make_span(axes_));
           // Reset per-sample values, but don't postprocess
           mean.Run(true, false);
-        });
+        }, volume(in_view[i].shape));
       }
-      tp.WaitForWork();
+      tp.RunAll();
       // Aggregate and postprocess now
       FoldMeans();
     }
 
     if (ShouldCalcStdDev()) {
       auto sample_mean = mean_view[0];
       for (int i = 0; i < nsamples; i++) {
-        tp.DoWorkWithID([&, i](int thread_idx) {
+        tp.AddWork([&, i](int thread_idx) {
           kernels::VarianceCPU<float, InputType> stddev;
           stddev.Setup(mutable_stddev[i], in_view[i], make_span(axes_), sample_mean);
           // Reset per-sample values, but don't postprocess
           stddev.Run(true, false);
-        });
+        }, volume(in_view[i].shape));
```

Comment:
Maybe you can add a method that would do the same as `tensor_size` to shape here as well?